hoot-core/src/main/cpp/hoot/core/io/ApiDbReader.h

```diff
@@ -102,6 +143,17 @@ class ApiDbReader : public Boundable
   static bool _isValidBounds(const geos::geom::Envelope& bounds);
   bool _hasBounds();
 
+private:
+
+  void _read(OsmMapPtr map, const ElementType& elementType);
+
+  boost::shared_ptr<Element> _getElementUsingIterator();
+
+  /**
+   * Converts a query result to an OSM element
+   */
+  boost::shared_ptr<Element> _resultToElement(QSqlQuery& resultIterator,
```

Comment:
Add a comment here that this function call will advance resultIterator, THEN read the element.